client/lib/titan/is-subscription-suspended.ts

```diff
@@ -0,0 +1,10 @@
+import type { ResponseDomain } from 'calypso/lib/domains/types';
+import type { SiteDomain } from 'calypso/state/sites/domains/types';
+
+export function isSubscriptionSuspended(
+	domain: ResponseDomain | SiteDomain | undefined
+): boolean {
+	const subscriptionStatus = domain?.titanMailSubscription?.status ?? '';
+
+	return subscriptionStatus === 'suspended';
+}
```

Comment:
I think we could probably refer to Titan in the name of this selector like we do for other selectors, e.g. `isTitanSubscriptionSuspended`.